include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -2148,43 +2149,50 @@ class FEEvaluationAccess<1, 1, Number, is_face, VectorizedArrayType>
  * points) as a template parameter makes things more complicated in codes
  * where different polynomial degrees should be considered, e.g. in
  * application codes where the polynomial degree is given through an input
- * file. The recommended approach for good performance is to create different
- * cell functions, possibly through different operator classes that are
- * derived from a common base class with virtual functions to access the
- * operator evaluation. The program then keeps only a pointer to the common
- * base class after initializing templated derived classes with fixed
- * polynomial degree that are selected from the detected polynomial
- * degree. This approach requires a-priori knowledge of the range of valid
- * degrees and can lead to rather long compile times in programs with many
- * apply functions.
- *
- * A flexible choice of the polynomial degree based on the information in the
- * element passed to the initialization is also supported by FEEvaluation,
- * even though it runs two to three times more slowly. For this, set the
- * template parameter for the polynomial degree to -1 (and choose an arbitrary
- * number for the number of quadrature points), which switches to the other
- * code path. Internally, an evaluator object with template specialization for
- * -1 is invoked that runs according to run-time bounds, whereas the default
- * case with compile-time bounds given through fe_degree selects another
- * template class without compromising efficiency.
+ * file. The second usage model is to rely on pre-compiled code for polynomial
+ * degrees. While a user code can use different functions for the cells (that
+ * get e.g. invoked by some dynamic dispatch mechanism for the various degree
+ * templates), deal.II also supports usage of this class based on the
+ * information in the element passed to the initialization. For this usage
+ * model, set the template parameter for the polynomial degree to -1 and
+ * choose an arbitrary number for the number of quadrature points. That code
```

Comment:
> arbitrary number for the number of quadrature points

The value is not actually used, is it? I guess we cannot force it to be `-1` due to backwards compatibility.

Reply:
No, it is not used, but also it does not matter I believe because we replace the value here:
https://github.com/dealii/dealii/blob/1b5806b8e213f33e1773d7556a93f1efb618bdba/include/deal.II/matrix_free/evaluation_template_factory.templates.h#L55
If I remember exactly we also did not say explicitly that one needed the combination `-1,0` before, so I would think it does not matter. Or was there anything particular you had in mind that works in one case but not the other?

Reply:
Thanks. No, I think it might give somehow to a user the wrong impression if `n_quadrature_points` is not forced to `-1` and he/she sets the values to `>0` and gets the *wrong* (not expected) result.

But this is not that important for this PR. I am happy with the PR. I let @tjhei merge!